### ISSUE TITLE
How variables are supported in maily editor?

### ISSUE THREAD

**Issue by chavda-bhavik** [arikchakma/maily.to#16]
Amazing project and idea team. I'm planning to build and add a similar kind of editor to my project. Everything looks sorted, just confused about how `variables` are supported in the editor.

Of course not stopping me from building an email editor like `maily` but curious to know how you have planned to support `variables`.

Thank you.

**Comment by arikchakma** [comment]
I built this custom [Variable Node](https://github.com/arikchakma/maily.to/blob/main/components/editor/nodes/variable.tsx) extending the `@tiptap/extension-mention` plugin. And in the renderer, I am just rendering it as [{{variableId}}](https://github.com/arikchakma/maily.to/blob/3a8dde8cfc4b78bb3c1f39d8f853ed9e213d4958/components/editor/utils/email.ts#L410)

**Comment by chavda-bhavik** [comment]
Got it, Arik. We can add variable with @ symbol.

If I want to show a menu with a list of variables available when @ is written, then what would be a good way to do it?

**Comment by arikchakma** [comment]
I am using `tippy.js` for it, you can also use it.

**Comment by chavda-bhavik** [comment]
Got it. Will check it out. Thanks Arik. Awesome project. 👍🏼

**Comment by vrabota** [comment]
@arikchakma Can we use variables for subject as well?

**Comment by vrabota** [comment]
Your subject here is `Test Components`

**Comment by vrabota** [comment]
What is the difference here between Subject and Preview header?
Is that subject will be send to email provider as well? or you use it just internally?

One more question is related how we set values for this variables from backend, for ex: we have variable `{{user}}` and from our backend and database when we need to send the email we want to set it to `Joe`.
Or we just set at template level `{{user}}` and provider like Resend should handle this logic to convert it?
